src/main/java/seedu/address/model/contact/NameContainsKeywordsPredicate.java

```diff
@@ -9,23 +9,28 @@
  * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
  */
 public class NameContainsKeywordsPredicate implements Predicate<Contact> {
-    private final List<String> keywords;
+    private final Set<Name> nameKeywords;
 
-    public NameContainsKeywordsPredicate(List<String> keywords) {
-        this.keywords = keywords;
+    public NameContainsKeywordsPredicate(Set<Name> nameKeywords) {
+        this.nameKeywords = nameKeywords;
     }
 
     @Override
     public boolean test(Contact person) {
-        return keywords.stream()
-                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getName().fullName, keyword));
+        if (nameKeywords.isEmpty()) {
+            return true;
+        }
+
+        return nameKeywords.stream()
+                .anyMatch(nameKeyword ->
+                        StringUtil.containsWordIgnoreCase(person.getName().fullName, nameKeyword.fullName));
```

Comment:
Using `containsWordIgnoreCase` here gives `IllegalArgumentException` and `InvocationTargetException` when the user enters the command `findc n/Alex Yeoh` (or any name that has a space). It might be better to use `containsPhraseIgnoreCase` instead to allow for spaces in the search term.

Same for `ContactTagContainsKeywordsPredicate`